render/render.go

```diff
@@ -142,16 +227,33 @@ func edgeMetadata(t report.Topology, mapFunc MapFunc, srcRenderableID, dstRender
 			continue
 		}
 		if src != report.TheInternet {
-			mapped, _ := mapFunc(t.NodeMetadatas[src])
+			mapped, _ := m.Mapper(t.NodeMetadatas[src])
 			src = mapped.ID
 		}
 		if dst != report.TheInternet {
-			mapped, _ := mapFunc(t.NodeMetadatas[dst])
+			mapped, _ := m.Mapper(t.NodeMetadatas[dst])
 			dst = mapped.ID
 		}
 		if src == srcRenderableID && dst == dstRenderableID {
 			metadata.Flatten(edgeMeta)
 		}
 	}
-	return metadata
+	return metadata.Transform()
+}
+
+// Render produces a set of RenderableNodes given a Report
+func (f FilterUnconnected) Render(rpt report.Report) RenderableNodes {
+	input := f.Renderer.Render(rpt)
+	output := RenderableNodes{}
+	for id, node := range input {
+		if len(node.Adjacency) == 0 {
+			continue
+		}
+
+		output[id] = node
+		for _, id := range node.Adjacency {
+			output[id] = input[id]
+		}
+	}
+	return output
 }
```

Comment:
This was me being defensive; AFAICT there is nothing stopping symmetric edges (ie A->B without B->A).  In that case I wanted to be sure not to drop B.

If we want to ensure symmetric edges, I'll add a check to report.Validate.